spot_wrapper/wrapper.py

```diff
@@ -722,234 +466,228 @@ def __init__(
             self._valid = False
             return
 
-        if self._robot:
-            # Clients
-            self._logger.info("Creating clients...")
-            initialised = False
-            while not initialised:
+        if not self._robot:
+            self._logger.error("Failed to create robot object")
+            self._valid = False
+            return
+
+        # Clients
+        self._logger.info("Creating clients...")
+        initialised = False
+        while not initialised:
```

Comment:
I'm not sure I follow. I used a while loop here as part of the startup procedure to make it so that it's not necessary to wait for the robot to be started up fully. If it isn't started, some of these services will not be constructed and thus cause exceptions, which we want to ignore because we know that a good startup should not result in any exceptions. The while loop exits when everything initialises correctly.

Unless you mean that we should limit the number of attempts?